plugins/org.jboss.tools.openshift.cdk.server/src/org/jboss/tools/openshift/cdk/server/core/internal/CDKConstantUtility.java

```diff
@@ -10,29 +10,143 @@
  ******************************************************************************/ 
 package org.jboss.tools.openshift.cdk.server.core.internal;
 
-import org.eclipse.debug.core.ILaunchConfiguration;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Map;
+
 import org.eclipse.core.runtime.CoreException;
 import org.eclipse.core.runtime.NullProgressMonitor;
+import org.eclipse.core.runtime.Platform;
+import org.eclipse.debug.core.ILaunchConfiguration;
 import org.eclipse.wst.server.core.IServer;
 import org.jboss.tools.openshift.cdk.server.core.internal.adapter.controllers.IExternalLaunchConstants;
 
-// TODO - allow customization of this location
 public class CDKConstantUtility {
 	private static final String VAGRANT_LOCATION_LINUX = "/usr/bin/vagrant";
-
+	
+	// Seems weird?  See https://github.com/mitchellh/vagrant/issues/1652
+	private static final String VAGRANT_LOCATION_WINDOWS = "C:\\HashiCorp\\Vagrant\\bin\\vagrant.exe";
+	
+	// Linux command to find vagrant full path
+	private static final String LINUX_WHICH_VAGRANT = "which vagrant";
+	
+	// Windows command to find vagrant
+	private static final String WINDOWS_WHERE_VAGRANT = "where vagrant";
+	
+	
+	private static String discoveredVagrantLocation = null;
+	private static boolean searchFailed = false;
+	
+	
 	public static String getVagrantLocation() {
-		return VAGRANT_LOCATION_LINUX;
+		return findVagrantLocation();
```

Comment:
im wondering if we are better of launching this similar to what @gercan and @ibuziuk been doing for npm and cordova cli tools.

i.e. call out via bash or cmd.exe making it up to the user to have it properly configured in PATH